sklearn/multioutput.py

```diff
@@ -660,7 +660,9 @@ def _log_message(self, *, estimator_idx, n_estimators, processing_msg):
     def _get_predictions(self, X, *, output_method):
         """Get predictions for each model in the chain."""
         check_is_fitted(self)
-        X = validate_data(self, X, accept_sparse=True, reset=False)
+        X = validate_data(
+            self, X, accept_sparse=True, reset=False, ensure_all_finite=False
```

Comment:
With a new look, I think that we should leverage the tag here:

```suggestion
            self,
            X,
            accept_sparse=True,
            reset=False,
            ensure_all_finite=get_tags(self.base_estimator).input_tags.allow_nan,
```

Reply:
We should make sure that `__sklearn_tags__` is defined that that the `allow_nan` tags delagate to the tag of `self.base_estimator`.

Reply:
Something like:

```python
def def __sklearn_tags__(self):
    tags = super().__sklearn_tags__()
    # FIXME
    tags._skip_test = True
    tags.input_tags.allow_nan = get_tags(self.base_estimator).input_tags.allow_nan
    tags.target_tags.single_output = False
    tags.target_tags.multi_output = True
    return tags

Reply:
If we use `ensure_all_finite=not get_tags(self).input_tags.allow_nan` then we won't ensure there are no `inf` values in data. For example it will be relevant for `DecisionTreeRegressor`. It has `tags.input_tags.allow_nan is True`, but does not allow `inf`. Thus, exception will be raised not in chain, but in estimator.

Also we can't do something like `ensure_all_finite='allow_nan' if get_tags(self).input_tags.allow_nan else True`, because of `HistGradientBoostingRegressor`, it allows `inf` as well as `nan`.

Maybe `allow_nan` tag should be compliant with `ensure_all_finite` parameter.
Correct me if I am wrong